extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -0,0 +1,152 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { IGenerateScriptSettings, IPublishSettings } from '../models/IPublishSettings';
+import { Project } from '../models/project';
+import { promptForPublishProfile } from './publishDatabaseDialog';
+
+/**
+ * Create flow for Publishing a database using only VS Code-native APIs such as QuickPick
+ */
+export async function launchPublishDatabaseQuickpick(project: Project): Promise<void> {
+
+	// 1. Select publish settings file (optional)
+	// TODO: Hook up to dacfx service
```

Comment:
Do you have an issue/task for this effort that you can add to the TODOs to help make sure we don't miss any of them?

Reply:
Nah, since I'm going through these so quickly I don't really see the value in spending time maintaining a list. I'll be sure to do a search for TODO's once I've finished though to make sure I don't miss any =)